Libraries/LLM/Evaluate.swift

```diff
@@ -218,6 +305,12 @@ public func generate(
     let now = Date.timeIntervalSinceReferenceDate
     let generateTime = now - start
 
+    // TokenIterator uses `asyncEval()` to keep the pipeline full.  If the caller
+    // exits the program right away, those tasks will still be executing and will
+    // hit assertions as the mlx scheduler is torn down.  Evaluate something trivial
+    // to synchronize with the scheduler.
+    eval(MLXArray(0))
```

Comment:
After the fix of eval to simply wait if all arrays are scheduled, this will not work as this will become a noop.

We have two options:
1. Provide a stream synchronize
2. Make this a `eval(MLXArray(0) + MLXArray(0))`

Reply:
Ah, interesting!  It looks like python has:

```python
  m.def(
      "synchronize",
      [](const std::optional<Stream>& s) {
        s ? synchronize(s.value()) : synchronize();
      },
      "stream"_a = nb::none(),
```

and we have the required code on mlx-c:

```c
/**
 * Synchronize with the provided stream.
 */
void mlx_synchronize(mlx_stream stream);
```

Reply:
Oh, in fact there is already API for it :-)

Reply:
Thanks for spotting this -- fixed!